From 08e0522fe676356be1ab49598d0aa73ec18d3c8f Mon Sep 17 00:00:00 2001
From: Miaohe Lin <linmiaohe@huawei.com>
Date: Wed, 16 Mar 2022 16:15:28 +0800
Subject: [PATCH 026/921] [Backport] mm/damon: minor cleanup for damon_pa_young

mainline inclusion
from mainline-v6.2-rc6
commit 2a3c4bce3edb0d54983384aa8a88c0da330638f4
category: feature
bugzilla: 188442

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=2a3c4bce3edb0d54983384aa8a88c0da330638f4

--------------------------------

if need_lock is true but folio_trylock fails, we should return false
instead of NULL to match the return value type exactly. No functional
change intended.

Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>

Conflicts:
	mm/damon/paddr.c

Signed-off-by: Ze Zuo <zuoze1@huawei.com>
---
 mm/damon/paddr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/paddr.c b/mm/damon/paddr.c
index 5e8244f65a1a..a6c5b24fba7a 100644
--- a/mm/damon/paddr.c
+++ b/mm/damon/paddr.c
@@ -156,7 +156,7 @@ static bool damon_pa_young(unsigned long paddr, unsigned long *page_sz)
 	need_lock = !PageAnon(page) || PageKsm(page);
 	if (need_lock && !trylock_page(page)) {
 		put_page(page);
-		return NULL;
+		return false;
 	}
 
 	rmap_walk(page, &rwc);
-- 
2.17.1

